lib/solidus_paypal_commerce_platform/engine.rb

```diff
@@ -15,10 +15,17 @@ class Engine < Rails::Engine
     engine_name 'solidus_paypal_commerce_platform'
 
     initializer "solidus_paypal_commerce_platform.add_payment_method", after: "spree.register.payment_methods" do |app|
-      app.config.spree.payment_methods << SolidusPaypalCommercePlatform::PaymentMethod
-      SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types << :paypal_select
-      Spree::PermittedAttributes.source_attributes.concat [:paypal_order_id, :authorization_id,
-                                                           :paypal_email, :paypal_funding_source]
+      app.config.to_prepare do
+        app.config.spree.payment_methods << SolidusPaypalCommercePlatform::PaymentMethod
+
+        unless SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types.include?(:paypal_select)
+          SolidusPaypalCommercePlatform::PaymentMethod.allowed_admin_form_preference_types << :paypal_select
+        end
+
+        old_source_attributes = ::Spree::PermittedAttributes.source_attributes
+        new_source_attributes = [:paypal_order_id, :authorization_id, :paypal_email, :paypal_funding_source]
+        ::Spree::PermittedAttributes.source_attributes.concat(new_source_attributes - old_source_attributes)
+      end
     end
 
     initializer "solidus_paypal_commerce_platform.add_wizard", after: "spree.register.payment_methods" do |app|
```

Comment:
There's no need to wrap the remaining initializers in `to_prepare` callbacks.

* `solidus_paypal_commerce_platform.add_wizard` only stores `"SolidusPaypalCommercePlatform::Wizard"` as a string in the `payment_setup_wizards` set. It constantizes the wizard to the actual class only during at runtime. As such, the initializer doesn't have to autoload the class on boot.
* `solidus_paypal_commerce_platform.set_pricing_options_class` - `SolidusPaypalCommercePlatform::PricingOptions` is not autoloaded because it is only returned by the `pricing_options_class` method during runtime.
* `solidus_paypal_commerce_platform.webhooks` - `SolidusPaypalCommercePlatform::WebhooksJob` is not autoloaded because it is only referred to when the webhook handler is called.

For all of these initializers, the external dependencies  (`Spree`, `SolidusWebhooks`) have already been loaded prior to the initializers.